### PR TITLE
[istio] fix dataplane upgrade

### PR DESCRIPTION
**Pull request by fl64**
## Description
Added check of istiod operation before controller starts upgrading required resources

## Why do we need it, and what problem does it solve?
Upgraded resources may receive old versions of sidecars because the required istiod is not ready

## What is the expected result?
Resources are not upgraded as long as istiod is out of service

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: istio
type: fix
summary: Added check of istiod operation before controller starts upgrading required resources.
impact_level: default
```

### CODE REVIEW

**Comment by apolovov** [comment]
The dataplane upgrade doc is in wrong place. Move it to examples, please and add add two links to the alert:
* to control-plane upgrade instruction
* to data-plane upgrade instruction

**Review by apolovov — COMMENTED** [review]
The dataplane upgrade doc is in wrong place. Move it to examples, please and add add two links to the alert:

to control-plane upgrade instruction
to data-plane upgrade instruction

**Review by apolovov — CHANGES_REQUESTED** [review]
The dataplane upgrade doc is in wrong place. Move it to examples, please and add add two links to the alert:

* to control-plane upgrade instruction
* to data-plane upgrade instruction

File: ee/modules/110-istio/docs/EXAMPLES.md [hunk 1]
```diff
@@ -642,3 +644,7 @@ kubectl get pods -A -o json | jq --arg revision "v1x13" \
   '.items[] | select(.metadata.annotations."sidecar.istio.io/status" // "{}" | fromjson | 
    .revision == $revision) | .metadata.namespace + "/" + .metadata.name'
 ```
+
+### Auto upgrade istio data-plane
```
Comment (z9r5) [inline ee/modules/110-istio/docs/EXAMPLES.md:648]:
```suggestion
### Auto upgrading istio data-plane
```

**Review by z9r5 — DISMISSED** [review]

File: ee/modules/110-istio/monitoring/prometheus-rules/dataplane.yaml [hunk 2]
```diff
@@ -159,6 +159,7 @@
         ```
         max by (namespace, dataplane_pod) (d8_istio_dataplane_metadata{version="{{$labels.full_version}}"})
         ```
+        Also consider automating the istio data-plane update described in the documentation: https://deckhouse.io/documentation/v1/modules/110-istio/examples.html#upgrading-istio
```
Comment (apolovov) [inline ee/modules/110-istio/monitoring/prometheus-rules/dataplane.yaml:162]:
```suggestion
        Also consider using the automatic istio data-plane update described in the documentation: https://deckhouse.io/documentation/v1/modules/110-istio/examples.html#upgrading-istio
```

**Review by apolovov — DISMISSED** [review]

**Review by apolovov — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Pull request merged**